pkg/tsdb/sqleng/resample.go

```diff
@@ -110,17 +110,15 @@ func resample(f *data.Frame, qm dataQueryModel) (*data.Frame, error) {
 				return f, fmt.Errorf("time point is nil")
 			}
 
-			if t.(time.Time).After(currentTime) {
-				nextTime := currentTime.Add(qm.Interval)
-				if t.(time.Time).Before(nextTime) {
+			previousTime := currentTime.Add(-qm.Interval)
```

Comment:
![mind blown](https://media.giphy.com/media/fnpy7E5Sm1LSOvHvu8/giphy.gif)

Can we add a comment to this? 😅